### PR TITLE
Add facades and code coverage report

### PR DESCRIPTION
**Pull request by idmontie**

### CODE REVIEW

File: src/utilities/json-utilities.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Nectary\Utilities;
+
+/**
+ * Helper utility functions for dealing with JSON
+ */
+class Json_Utilities {
+  public static function get( $json, $path ) {
```
Comment (rraub) [inline src/utilities/json-utilities.php:19]: Could we add an example in a comment for the 'get' function?

File: src/utilities/json-utilities.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        $current = $current[ $part ];  
+      } else {
+        break;
+      }
+    }
+
+    return $current;
+  }
+
+  public static function get_or_default( $json, $path, $default ) {
```
Comment (rraub) [inline src/utilities/json-utilities.php]: Why a separate function? what abut a $default=null ?

Comment (idmontie) [inline src/utilities/json-utilities.php]: I was lazy :zzz:

File: src/utilities/json-utilities.php (hunk 1 above)
Comment (idmontie) [inline src/utilities/json-utilities.php:19]: Sure. There are also tests that document how it works.

File: tests/unit/json-utilities-test.php [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        }
+      }
+    ";
+
+    $json = json_decode( $raw, true );
+
+    $value = Json_Utilities::get_or_default( $json, 'key1.key2.keyNope', '???' );
+    $this->assertEquals( '???', $value );
+  }
+}
```
Comment (rraub) [inline tests/unit/json-utilities-test.php:171]: What about the multiple elements case:

```
{
    \"key1\": {
        \"key2\": [
            \"value1\",
            \"value2\"
        ]
    }
}
"key1.key2.0"
```

File: src/utilities/string-utilities.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+   * Alias
+   */
+  function ends_with( $haystack, $needle ) {
+    $haystack_length = strlen( $haystack );
+    $needle_length   = strlen( $needle );
+    if ( $needle_length > $haystack_length )
+      return false;
+    return substr_compare( $haystack, $needle, -$needle_length ) === 0;
+  }
+}
```
Comment (rraub) [inline src/utilities/string-utilities.php:23]: What about adding the slugify function or should we wait until we need it?

Comment (idmontie) [inline src/utilities/string-utilities.php:23]: I'm _trying_ (but still failing) to write the tests for functions as I move them over. Just haven't gotten to that one yet.

File: tests/unit/json-utilities-test.php (hunk 3 above)
Comment (idmontie) [inline tests/unit/json-utilities-test.php:171]: :+1:

**Comment by rraub** [comment]
:+1:

**Pull request merged**